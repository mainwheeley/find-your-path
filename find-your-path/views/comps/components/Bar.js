import React from 'react';
import { StyleSheet, Dimensions, Text, View, Button, Alert, FlatList } from 'react-native';

export default class Bar extends React.Component {
   render() {
    return (

	<View style={styles.bar}>

	    <View style={[styles.barItem, styles.barseparator]}>
		<Text style={styles.barTop} onPress={() => this.props.navigation.navigate("FavoriteRoutes")}>Favorite Workouts</Text>
		<Text style={styles.barBottom}></Text>
	    </View>

	    <View style={[styles.barItem, styles.barseparator]}>
		<Text style={styles.barTop}>10</Text>
		<Text style={styles.barBottom}>Following</Text>
	    </View>

	</View>

    );
  }
}

const styles = StyleSheet.create({
    bar: {
	borderColor: '#fff',
	borderTopWidth: 4,
	height: Dimensions.get('window').height / 10,
	backgroundColor: '#8BC34A',
	flexDirection: 'row',
    },
    barSeparator: {
	borderRightWidth: 4
    },
    barTop: {
	color: '#fff',
	fontSize: 16,
	fontWeight: 'bold',
	fontStyle: 'italic',
    },
    barItem: {
	flex: 1,
	padding: 18,
	alignItems: 'center',
    },
    barBottom: {
	color: '#000',
	fontSize: 14,
	fontWeight: 'bold',
    },
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
