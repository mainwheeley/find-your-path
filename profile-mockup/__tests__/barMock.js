import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Bar from '../comps/components/Bar';
import { shallow } from 'enzyme';
describe('Profile Bar', function() {
    it('renders correctly', () => {
	const tree = shallow(<Bar />)
	expect(tree).toMatchSnapshot();
    });
    it('length 1', function() {
	const tree = shallow(<Bar />)
	expect(tree.length).toBe(1);
    });
});
