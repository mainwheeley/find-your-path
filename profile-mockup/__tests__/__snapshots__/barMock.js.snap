// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Bar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
      Object {
        "backgroundColor": "#8BC34A",
        "borderColor": "#fff",
        "borderTopWidth": 4,
        "flexDirection": "row",
        "height": 133.4,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 18,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }
      >
        10
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Followers
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 18,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontStyle": "italic",
            "fontWeight": "bold",
          }
        }
      >
        10
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Following
      </Text>
    </View>
  </View>,
  "nodes": Array [
    <View
      style={
        Object {
          "backgroundColor": "#8BC34A",
          "borderColor": "#fff",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "padding": 18,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          10
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Followers
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "padding": 18,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          10
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Following
        </Text>
      </View>
    </View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Bar />,
    "_instance": Bar {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
        Object {
          "backgroundColor": "#8BC34A",
          "borderColor": "#fff",
          "borderTopWidth": 4,
          "flexDirection": "row",
          "height": 133.4,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "padding": 18,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          10
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Followers
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "padding": 18,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontStyle": "italic",
              "fontWeight": "bold",
            }
          }
        >
          10
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Following
        </Text>
      </View>
    </View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Bar />,
}
`;
